### DISCUSSION TITLE
Display Image File Name from Image Slideshow - Possible in Advanced Scene Switcher?

### DISCUSSION THREAD

**Discussion by ProgramDeez** [WarmUpTill/SceneSwitcher#1131] [Q&A] [answered]
Trying to find a way to display the file name of the image that is currently being shown from within an 'Image Slideshow' source. The idea would be that as the image changes in the slideshow, the file name would change to that of the new image also. 

Hoping to get this to work for an image slideshow that is pulling from a directory, rather than a static list of images entered into the 'Image Files' field, which I believe complicates obtaining the file name of the current image being displayed.

Not sure if it is possible, but was curious if there is a way for Advance Scene Switcher to perform this type of function. If so, I would greatly appreciate any insight as to how to best go about implementing this. Thanks for any help in advance!

**Comment by WarmUpTill** [comment, marked as answer]
Just replied on the OBS forums but for completeness I will add the reply here as well:

While this should already be possible with the current version by using the "Slide Show" condition type it should be a lot easier in the following test build:
https://github.com/WarmUpTill/SceneSwitcher/actions/runs/9522815797?pr=1133
It adds the option to directly query the file name of an image slide show and adds support for the "slideshow_v2" source type.
Unfortunately, you will have to be logged into GitHub to be able to download this build.
Let me know if that should be a problem for you.

When you are using this build you should be able to achieve what you are looking for by setting up a macro similar to this example:
![](https://obsproject.com/forum/attachments/1718402785132-png.104608/)

It boils down to "if the currently displayed slide of the source "Image Slide Show" changed assign the file name of the file being displayed to the variable "Slide Show File" and set the "Text" settings value of the "Text(GDI+)" source to the value of that variable".

Let me know if you run into any issues with that build or if you have any questions!
Hope this helped! :)

**Comment by ProgramDeez** [comment]
Thanks so much for your reply! Incredibly helpful! I mostly have been able to replicate your example, however am running into an issue when it comes to selecting for the 'Action Type' in the 'Variable' section. Under 'Variable' > 'Set to macro property' > 'Slide Show File' > '--sele...' I am unable to access any action types in the dropdown or being able to write anything in this field.

I did try to implement the build you shared, but I am unsure that I properly integrated, as it seems like this may be the cause of the issue. Please let me know if that is the case or if you believe there is another way I could go about resolving. Appreciate the assistance again! 

My replicated example below for reference: 
![OBSIssue](https://github.com/WarmUpTill/SceneSwitcher/assets/87685099/aa0f6dc3-023f-45e2-b8d6-88b476f604a2)

**Reply by WarmUpTill** [reply]
Can you double check that the test build is installed?
I would expect the version number to say the following:
![grafik](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/c37c5cc5-719f-43a9-859e-0361987e6440)

If that is already the version number you are currently using, does it help to remove the condition and re-add it?

**Reply by ProgramDeez** [reply]
Got it working! That was exactly the issue, did not have the test build installed properly.

Appreciate you taking the time to help with this. You are awesome and Adv-SS is absolutely incredible!
